tests/aws/services/route53resolver/test_route53resolver.py

```diff
@@ -721,3 +722,221 @@ def test_list_firewall_domain_lists(self, cleanups, snapshot, aws_client):
+            Name=firewall_rule_group_name
+        )
+        snapshot.match("create-firewall-rule-group", rule_group_response)
+        cleanups.append(
+            lambda: aws_client.route53resolver.delete_firewall_rule_group(
+                FirewallRuleGroupId=rule_group_response["FirewallRuleGroup"]["Id"]
+            )
+        )
+
+        # Add 4 domain lists
+        domain_list_response_1 = aws_client.route53resolver.create_firewall_domain_list(
+            Name=f"fw-domain-list-{short_uid()}"
+        )
+        snapshot.match("create-firewall-domain-list-1", domain_list_response_1)
+        cleanups.append(
+            lambda: aws_client.route53resolver.delete_firewall_domain_list(
+                FirewallDomainListId=domain_list_response_1["FirewallDomainList"]["Id"]
+            )
+        )
+
+        domain_list_response_2 = aws_client.route53resolver.create_firewall_domain_list(
+            Name=f"fw-domain-list-{short_uid()}"
+        )
+        snapshot.match("create-firewall-domain-list-2", domain_list_response_2)
+        cleanups.append(
+            lambda: aws_client.route53resolver.delete_firewall_domain_list(
+                FirewallDomainListId=domain_list_response_2["FirewallDomainList"]["Id"]
+            )
+        )
+        domain_list_response_3 = aws_client.route53resolver.create_firewall_domain_list(
+            Name=f"fw-domain-list-{short_uid()}"
+        )
+        snapshot.match("create-firewall-domain-list-3", domain_list_response_3)
+        cleanups.append(
+            lambda: aws_client.route53resolver.delete_firewall_domain_list(
+                FirewallDomainListId=domain_list_response_3["FirewallDomainList"]["Id"]
+            )
+        )
+        domain_list_response_4 = aws_client.route53resolver.create_firewall_domain_list(
+            Name=f"fw-domain-list-{short_uid()}"
+        )
+        snapshot.match("create-firewall-domain-list-4", domain_list_response_4)
+        cleanups.append(
+            lambda: aws_client.route53resolver.delete_firewall_domain_list(
+                FirewallDomainListId=domain_list_response_4["FirewallDomainList"]["Id"]
+            )
+        )
+
+        # Add 4 FW rules
+        rule_response_1 = aws_client.route53resolver.create_firewall_rule(
+            FirewallRuleGroupId=rule_group_response["FirewallRuleGroup"]["Id"],
+            FirewallDomainListId=domain_list_response_1["FirewallDomainList"]["Id"],
+            Priority=1,
+            Action=Action.ALLOW,
+            Name=f"rule-name-{short_uid()}",
+        )
+        snapshot.match("create-firewall-rule-1", rule_response_1)
+        cleanups.append(
+            lambda: aws_client.route53resolver.delete_firewall_rule(
+                FirewallRuleGroupId=rule_group_response["FirewallRuleGroup"]["Id"],
+                FirewallDomainListId=domain_list_response_1["FirewallDomainList"]["Id"],
+            )
+        )
```

Comment:
As this piece of code gets duplicated a few times, I would suggest to create a separate utility fixture to create the firewall rule and pass the arguments for `Priority` and `Action` which would additionally also cleanup the resources. 

Let me know if you require more inputs here, happy to help (you can find several references for this in the core repository itself 🙂)

Reply:
Created a new fixture for this. Thanks!